File: programmers/가장 큰 수.py
"""
코딩테스트 연습 > 정렬
"""


def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*3, reverse=True)
    numbers = list(map(str, map(int, numbers)))
    res = ''.join(numbers)
    res = str(int(res))

    return res

File: programmers/가장 큰 수.kt
class Solution {
    fun solution(numbers: IntArray): String {
        return if (numbers.toList().maxOrNull() == 0) {
            "0"
        } else {
            numbers
                .map{ it.toString() }
                .sortedWith(compareBy({it.repeat(4)}))
                .reversed()
                .joinToString("")
        }
    }
}
